src/main/java/com/minecolonies/coremod/colony/TriggerColonyAchievements.java

```diff
@@ -0,0 +1,246 @@
+                colony.triggerAchievement(ModAchievements.achievementCatch100Fish);
+                break;
+            case TAG_FARMER_WHEAT:
+                colony.triggerAchievement(ModAchievements.achievementHarvest100Wheat);
+                break;
+            case TAG_FARMER_POTATOES:
+                colony.triggerAchievement(ModAchievements.achievementHarvest100Potatoes);
+                break;
+            case TAG_FARMER_CARROTS:
+                colony.triggerAchievement(ModAchievements.achievementHarvest100Carrots);
+                break;
+            case TAG_LUMBERJACK_SAPLINGS:
+                colony.triggerAchievement(ModAchievements.achievementPlant100Saplings);
+                break;
+            case TAG_LUMBERJACK_TREES:
+                colony.triggerAchievement(ModAchievements.achievementFell100Trees);
+                break;
+            default:
+                break;
+        }
+    }
+
+    /**
+     *Trigger fifth achievement.
+     * @param statistic the statistic.
+     * @param colony the colony.
+     */
+    public static void triggerFourthAchievement(@NotNull String statistic, @NotNull Colony colony)
+    {
+        switch (statistic)
+        {
+            case TAG_GUARD_MOBS:
+                colony.triggerAchievement(ModAchievements.achievementKill500Mobs);
+                break;
+            case TAG_MINER_ORES:
+                colony.triggerAchievement(ModAchievements.achievementMine500Ores);
+                break;
+            case TAG_MINER_DIAMONDS:
+                colony.triggerAchievement(ModAchievements.achievementMine500Diamonds);
+                break;
+            case TAG_BUILDER_HUTS:
+                colony.triggerAchievement(ModAchievements.achievementBuild500Huts);
+                break;
+            case TAG_FISHERMAN_FISH:
+                colony.triggerAchievement(ModAchievements.achievementCatch500Fish);
+                break;
+            case TAG_FARMER_WHEAT:
+                colony.triggerAchievement(ModAchievements.achievementHarvest500Wheat);
+                break;
+            case TAG_FARMER_POTATOES:
+                colony.triggerAchievement(ModAchievements.achievementHarvest500Potatoes);
+                break;
+            case TAG_FARMER_CARROTS:
+                colony.triggerAchievement(ModAchievements.achievementHarvest500Carrots);
+                break;
+            case TAG_LUMBERJACK_SAPLINGS:
+                colony.triggerAchievement(ModAchievements.achievementPlant500Saplings);
+                break;
+            case TAG_LUMBERJACK_TREES:
+                colony.triggerAchievement(ModAchievements.achievementFell500Trees);
+                break;
+            default:
+                break;
+        }
+    }
+
+    /**
+     *Trigger fifth achievement.
+     * @param statistic the statistic.
+     * @param colony the colony.
+     */
+    public static void triggerFifthAchievement(@NotNull String statistic, @NotNull Colony colony)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') The Cyclomatic Complexity of this method "triggerFifthAchievement" is 12 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)